### PR TITLE
[Merged by Bors] - fix(stream_model): prevent collisions between namespaces

### PR DESCRIPTION
**Pull request by nacardin**

### CODE REVIEW

File: crates/fluvio-stream-model/src/store/k8.rs [hunk 1]
```diff
@@ -163,11 +163,13 @@ pub trait K8ExtendedSpec: Spec {
 
     fn convert_from_k8(
         k8_obj: K8Obj<Self::K8Spec>,
+        multi_namespace_context: bool,
     ) -> Result<MetadataStoreObject<Self, K8MetaItem>, K8ConvertError<Self::K8Spec>>;
 }
 
 pub fn default_convert_from_k8<S>(
```
Comment (morenol) [inline crates/fluvio-stream-model/src/store/k8.rs:170]: I think that we should add some references to the way that we should use `fluvio_stream_dispatcher::store::StoreContext` when using multiple namespaces for all the methods that have `key: S::IndexKey` as a parameter. Maybe as docstrings?

**Review by morenol — APPROVED** [review]

File: crates/fluvio-stream-model/src/store/k8.rs (hunk 1 above)
Comment (nacardin) [reply crates/fluvio-stream-model/src/store/k8.rs:172]: added docstring to `default_convert_from_k8`

**Comment by sehz** [comment]
Can you add more description to this PR?

**Comment by sehz** [comment]
This is probably not fix since this is not bug fix

**Pull request closed without merging**